### ISSUE TITLE
'Robot Name' false positive tests

### ISSUE THREAD

**Issue by garveychan** [exercism/typescript#943]
## Issue

The [2nd](https://exercism.org/tracks/typescript/exercises/robot-name/solutions/jcsawyer123) and [3rd](https://exercism.org/tracks/typescript/exercises/robot-name/solutions/mangapi) ranked solutions for 'Robot Name' in Typescript are currently passing all unit tests even though (and primarily because) they are generating names that **_do not conform_** to the problem specification.

While the tests interpret these names as 'unique' because they do not exist within their respective Sets, repeating the regex check for each 'new name' function invocation exposes these strings as invalid robot names.

<img width="511" alt="Screenshot 2022-02-14 at 1 38 19 am" src="https://user-images.githubusercontent.com/79407762/153759138-2f0893e5-1ba1-47d3-a59c-e5ceb971dc7c.png">
<img width="511" alt="Screenshot 2022-02-14 at 1 33 08 am" src="https://user-images.githubusercontent.com/79407762/153759017-5de221d8-a458-4388-80ff-27ee09749d0a.png">

## Notes
- This issue may need to be addressed in other tracks as well.
- Implementing the regex check for each newly generated name adds about 10 seconds to even the most optimal (<500ms) solutions when checking all possible permutations. There may be a more performant approach for strengthening these tests.

<img width="511" alt="Screenshot 2022-02-14 at 1 42 20 am" src="https://user-images.githubusercontent.com/79407762/153759392-f3694aaf-d85f-48ce-aad6-a68dd22e1adb.png">

**Comment by garveychan** [comment]
Yep, that's right. My main concern with the regex quick fix is that it may distort the timeout validation when the solution is submitted online (I'm unfamiliar with the inner workings). I'll experiment with other options when I can find some more time.

**Comment by SleeplessByte** [comment]
Ah. Understood, thank you for that clarification. Want to post the regex test change you made? I can probably help you optimise the regex to make it faster.

#947 actually starts skipping the `all the names can be generated` test, so the timeout won't be that much of an issue anymore.

**Comment by garveychan** [comment]
I took the [existing regex test](https://github.com/exercism/typescript/blob/0e2558848129b1589b64a0dab8195a3134e5b411/exercises/practice/robot-name/robot-name.test.ts#L39) and put it into the loops where new robots were instantiated - on lines [68](https://github.com/exercism/typescript/blob/0e2558848129b1589b64a0dab8195a3134e5b411/exercises/practice/robot-name/robot-name.test.ts#L68) and [137](https://github.com/exercism/typescript/blob/0e2558848129b1589b64a0dab8195a3134e5b411/exercises/practice/robot-name/robot-name.test.ts#L137).

**Comment by SleeplessByte** [comment]
Sad. That's already pretty much as optimal as we can get. I'll figure out if there is a way to still test each and every name...

**Comment by SleeplessByte** [comment]
I have thought of a way to implement this, so I will.
